tasks/go.py

```diff
@@ -108,13 +109,20 @@ def deps(ctx, verbose=False):
     """
     Setup Go dependencies
     """
-
+    max_retry = 3
     print("downloading dependencies")
     with timed("go mod download"):
         verbosity = ' -x' if verbose else ''
         for mod in DEFAULT_MODULES.values():
             with ctx.cd(mod.full_path()):
-                ctx.run(f"go mod download{verbosity}")
+                for retry in range(max_retry):
+                    result = ctx.run(f"go mod download{verbosity}")
+                    if result.exited is None or result.exited > 0:
+                        wait = 10 ** (retry + 1)
+                        print(f"[{(retry + 1) / max_retry}] Failed downloading {mod.path}, retrying in {wait} seconds")
+                        sleep(wait)
```

Comment:
NIT: Was there any function that does a retry like this already (as in macos jobs) ? Maybe we can wrap this into a function

Reply:
yes I was thinking about a generic function decorator as well, but I was not sure how it will impact the method itself. It's maybe not a big refactor, that can take place later